pyccel/ast/core.py

```diff
@@ -177,7 +181,113 @@ def __init__(self, var):
         # Call the base class constructor with the parameters it needs
         super(AstFunctionResultError, self).__init__(msg)
 
+# Pow, Add, Mul need to inherite sympy.Boolean to be able to use them in a logical expression
 
+class Pow(sp_Pow, sp_Boolean, PyccelAstNode):
+
+    def _eval_subs(self, old, new):
+        args = self.args
+        args_ = [self.base._subs(old, new),self.exp._subs(old, new)]
+        args  = [args_[i] if args_[i] else args[i] for i in range(len(args))]
+        expr = Pow(args[0], args[1], evaluate=False)
+        return expr
+
+    def _eval_evalf(self,prec):
+        return sp_Pow(self.base,self.exp).evalf(prec)
+
+    @property
+    def rank(self):
+        # TODO: Use broadcasting rules to decide shape (https://docs.scipy.org/doc/numpy/user/basics.broadcasting.html)
+        return max(getattr(a,'rank',0) for a in self._args)
```

Comment:
There is already a Pow class defined in this file. Which Pow should we be using?